src/libtorch.cc

```diff
@@ -829,19 +846,21 @@ ModelInstanceState::ValidateOutputs()
     // Validate name
     std::string io_name;
     RETURN_IF_ERROR(io.MemberAsString("name", &io_name));
-    try {
-      int start_pos = io_name.find(deliminator);
-      if (start_pos == -1) {
-        throw std::invalid_argument("output must follow naming convention");
+    switch (naming_convention) {
+      case NamingConvention::NAMED_INDEX: {
+        int start_pos = io_name.find(deliminator);
+        if (start_pos == -1) {
+          throw std::invalid_argument("output must follow naming convention");
```

Comment:
Didn't this get validated already in GetNamingConvention() line 1340?

Reply:
Yes good catch